### PR TITLE
[Tabs] Display pretty source after reload

### PR DESCRIPTION
**Pull request by AnshulMalik**
Associated Issue: #3305 

### Summary of Changes

* Check if the selected the selected source was prettified, if yes, prettify the original source

### Test Plan

- [x] Prettify a source and reload debugger

### Screenshots/Videos (OPTIONAL)

t

### CODE REVIEW

File: src/actions/sources.js [hunk 1]
```diff
@@ -49,11 +49,23 @@ import type { ThunkArgs } from "./types";
...
 
-    if (pendingLocation && !!source.url && pendingLocation.url === source.url) {
+    if (!pendingLocation || !pendingLocation.hasOwnProperty("url")) {
+      return;
+    }
+
+    const { url: pendingURL } = pendingLocation;
+    const formattedPendingURL = pendingURL.replace(/:formatted$/, "");
+    if (!!source.url && formattedPendingURL === source.url) {
+      const isPendingLocationPretty = pendingURL.endsWith(":formatted");
```
Comment (jasonLaster) [inline src/actions/sources.js]: i think we have a helper for this `isPretty` or something

File: src/actions/sources.js [hunk 2]
```diff
@@ -49,11 +49,23 @@ import type { ThunkArgs } from "./types";
...
+
+      if (isPendingLocationPretty) {
+        return await dispatch(togglePrettyPrint(source.id));
```
Comment (jasonLaster) [inline src/actions/sources.js]: do we need to do `selectLocation` if we do this?

Comment (AnshulMalik) [reply src/actions/sources.js]: Nope, works out-of-the box

File: src/actions/sources.js (hunk 1 above)
Comment (AnshulMalik) [reply src/actions/sources.js]: Will find out :)

**Comment by AnshulMalik** [comment]
Hey, I can't figure out why the travis build is failing, 
What I can figure out from logs is `browser_dbg-expressions-error.js` is causing error.
But don't know where exactly. `help needed` :P

**Comment by jasonLaster** [comment]
Happy to help :)

**Comment by jasonLaster** [comment]
This PR is really clean!

* rebased
* made a couple of tweaks
* added a unit test because I think checkSelectedSource is complicated
* moved the mochitest to its own file because tests with reload often timeout. The mochitest is great! this is complicated enough where i'd expect we'd break it w/o a mochitest!

**Pull request merged**